services/ui-src/src/components/modals/AddEditProgramModal.tsx

```diff
@@ -83,6 +83,7 @@ export const AddEditProgramModal = ({
           ...dataToWrite.metadata,
           reportType,
           status: ReportStatus.NOT_STARTED,
+          isComplete: false,
```

Comment:
Question: Why is this necessary?

Reply:
Short answer: It's not necessary here, I'll move it. 

Long answer: It's easier to have the backend update an `isComplete` boolean rather than computing it to know if we can show submit buttons. This boolean could reasonably be set in the report create route, but I'd still like it to be initialized to `false` instead of left undefined when we create a report. This allows us to avoid optionals in typing the metadata interface. I'll move this (and setting the status) to the reports create route.

Reply:
Agreed that it makes sense to have it happen in route creation. Is this being stored alongside other report metadata, like the full completeness object? My concern is that they may somehow get out of sync with each other, where if we only ever update the completeness object and then calculate full completeness of it, you don't have that problem. 

What might be easiest is having a useEffect in ReportProvider calculate completeness and update `isComplete` anytime the completeness object changes. Alternatively, if the backend also needs access to it, you could have the fetch function perform the calculation and stitch `isComplete` into the metadata the same way [we currently combine metadata, fieldData and formTemplate](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/blob/e12110fa9ab9e2ad9f66d87e15b532816949c46e/services/app-api/handlers/reports/fetch.ts#L77-L81) before they get returned and set in ReportProvider.

Reply:
addressing under MDCT-2336